### PR TITLE
ui: use global state for applications data

### PR DESCRIPTION
**Pull request by Kenil27**
### Describe your changes:

Call applications API and set it in application state at root level

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AppContainer/AppContainer.tsx [hunk 1]
```diff
@@ -55,6 +67,7 @@ const AppContainer = () => {
     if (currentUser?.id) {
       fetchLimitConfig();
     }
+    fetchApplicationList();
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AppContainer/AppContainer.tsx:70]: @Kenil27 , I believe we should only fetch the applications if currentUser is available.

File: openmetadata-ui/src/main/resources/ui/src/hooks/useApplicationStore.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ import { EntityUnion } from '../components/Explore/ExplorePage.interface';
 import { AuthenticationConfiguration } from '../generated/configuration/authenticationConfiguration';
 import { AuthorizerConfiguration } from '../generated/configuration/authorizerConfiguration';
 import { UIThemePreference } from '../generated/configuration/uiThemePreference';
+import { App } from '../generated/entity/applications/app';
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/hooks/useApplicationStore.ts:20]:
```suggestion
import { App as Application } from '../generated/entity/applications/app';
```

File: openmetadata-ui/src/main/resources/ui/src/hooks/useApplicationStore.ts [hunk 3]
```diff
@@ -164,6 +166,9 @@ export const useApplicationStore = create<ApplicationStore>()(
       updateSearchCriteria: (criteria) => {
         set({ searchCriteria: criteria });
       },
+      setApplications: (applications: App[]) => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/hooks/useApplicationStore.ts:169]:
```suggestion
      setApplications: (applications: Application[]) => {
```

File: openmetadata-ui/src/main/resources/ui/src/interface/store.interface.ts [hunk 4]
```diff
@@ -55,6 +56,7 @@ export interface ApplicationStore
   searchCriteria: ExploreSearchIndex | '';
   theme: UIThemePreference['customTheme'];
   inlineAlertDetails?: InlineAlertProps;
+  applications: App[];
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/interface/store.interface.ts:59]:
```suggestion
  applications: Application[];
```

File: openmetadata-ui/src/main/resources/ui/src/interface/store.interface.ts [hunk 5]
```diff
@@ -82,6 +84,7 @@ export interface ApplicationStore
   removeRefreshToken: () => void;
   updateSearchCriteria: (criteria: ExploreSearchIndex | '') => void;
   trySilentSignIn: (forceLogout?: boolean) => void;
+  setApplications: (applications: App[]) => void;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/interface/store.interface.ts:87]:
```suggestion
  setApplications: (applications: Application[]) => void;
```

File: openmetadata-ui/src/main/resources/ui/src/components/AppContainer/AppContainer.tsx [hunk 6]
```diff
@@ -46,6 +47,17 @@ const AppContainer = () => {
...
+  const fetchApplicationList = useCallback(async () => {
+    try {
+      const { data } = await getApplicationList({
+        limit: 100,
+      });
+      setApplications(data);
+    } catch (error) {
+      // silent fail
+    }
+  }, []);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/AppContainer/AppContainer.tsx:59]: @Kenil27 why can't we update it from Application Provider itself?

**Review by chirag-madlani — DISMISSED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**